File: stack_traversal.cpp
// C++ program to traversal in an stack 
#include<stdio.h>
#include<stack>
#include<iostream>
using namespace std;
 
//Function to print the element in stack
void printStack(stack<int>& St)
{
    // Traverse the stack
    while(!St.empty()){
        //Print top Element
        cout<<St.top()<< ' ';
        //pop top element
        St.pop();
    }
}

int main()
{
    // Initialise stack
    stack<int> St;
    
    //Insert Element in stack
    St.push(4);
    St.push(3);
    St.push(2);
    St.push(1);
    
    //Print elements in stack
    printStack(St);
    return 0;
}

File: stack_traversal.java
// Traversal in an stack
import java.util.*;
class GFG{

// Function to print the element in stack
static void printStack(Stack<Integer> St)
{
	// Traverse the stack
	while (!St.isEmpty()) {

		// Print top element
		System.out.print(St.peek() +" ");

		// Pop top element
		St.pop();
	}
}

public static void main(String[] args)
{
	// Initialise stack
	Stack<Integer> St = new Stack<>() ;

	// Insert Element in stack
	St.add(4);
	St.add(3);
	St.add(2);
	St.add(1);

	// Print elements in stack
	printStack(St);
}
}
